From 3f6a5f41570c2f6feb5f7204286ef41739169b4e Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:24 -0400
Subject: [acpi] hotplug/pci: Do not clear event callback pointer for docks

Message-id: <1411129776-21656-50-git-send-email-prarit@redhat.com>
Patchwork-id: 96394
O-Subject: [RHEL7.1 PATCH BZ 1141846 49/61] ACPI / hotplug / PCI: Do not clear event callback pointer for docks
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit d7c7c0256b936901eb7a70fbc6d674a6bdfd463e
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Feb 21 01:08:42 2014 +0100

    ACPI / hotplug / PCI: Do not clear event callback pointer for docks

    After recent changes adding dock station handling to the ACPI hotplug
    core, it is not necessary to clear the .event() pointer in the
    ACPIPHP device hotplug context for dock stations any more, so don't
    do that.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 99e0717..ff8af5e 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -307,13 +307,6 @@ static acpi_status acpiphp_add_context(acpi_handle handle, u32 lvl, void *data,
  if (acpi_has_method(handle, "_STA"))
   newfunc->flags |= FUNC_HAS_STA;
 
- /*
-  * Dock stations' notify handler should be used for dock devices instead
-  * of the common one, so clear hp.event in their contexts.
-  */
- if (acpi_has_method(handle, "_DCK"))
-  context->hp.event = NULL;
-
  acpi_unlock_hp_context();
 
  /* search for objects that share the same slot */
-- 
1.7.1